### PR TITLE
Implement a working dev server

### PR DESCRIPTION
**Pull request by sansyrox**
Will not merge as Pyo3 is not compiling on Mac OS and am unable to test the implementation.

### CODE REVIEW

File: robyn/__init__.py [hunk 1]
```diff
@@ -15,7 +20,7 @@ def add_route(self, route_type, endpoint, handler):
 
     def start(self, port):
         print(f"Starting the server at port: {port}")
-        self.server.start(port)
+        self.server.start_dev_server(port)
```
Comment (sansyrox) [inline robyn/__init__.py:23]: Need to start based on flags.

**Comment by sansyrox** [comment]
@JackThomson2 , can you please have a look at the approach of this PR?

The implementation is clearly hacky needs improvement however I believe since it is a dev mode feature only and gets the hot reload to work it is a decent way.

What do you think?

**Comment by JackThomson2** [comment]
@sansyrox sure I'll try take a look this weekend if that's alright, busy week unfortunately

**Comment by sansyrox** [comment]
> sure I'll try take a look this weekend if that's alright, busy week unfortunately

@JackThomson2 , yep yep 🖖

File: robyn/__init__.py [hunk 2]
```diff
@@ -1,21 +1,66 @@
+# default imports
+import os
+import subprocess
+
+import argparse
 from .robyn import Server
 from asyncio import iscoroutinefunction
 from .responses import static_file, jsonify
 
+from watchdog.observers import Observer
```
Comment (sansyrox) [inline robyn/__init__.py:10]: need to include this as a dev dependency in pyproject.toml

**Review by sansyrox — COMMENTED** [review]
Dependencies issue has been fixed! 🔥 

Idk why pyo3 is failing again!

**Comment by sansyrox** [comment]
This works well on all machines. Something wrong with the pythoning.

**Pull request merged**